.github/workflows/build.yml

```diff
@@ -91,3 +129,47 @@ jobs:
         with:
           name: reports
           path: build/reports
+  deploy:
+    name: Deploy
+    needs: build
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    environment: ${{ needs.build.outputs.deploy_env }}
+    env:
+      THE_ENV: ${{ needs.build.outputs.deploy_env_lower }}
+      TARGET_ENV_UPPERCASE: ${{ needs.build.outputs.deploy_env }}
+      THE_VERSION: ${{ needs.build.outputs.version }}
+    if: |
+      github.ref == 'refs/heads/develop' ||
+      github.ref == 'refs/heads/main'    ||
+      startsWith(github.ref, 'refs/heads/release') ||
+      github.event_name == 'workflow_dispatch'
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-west-2
+          role-session-name: GitHubActions
+          aws-access-key-id: ${{ secrets[vars.AWS_ACCESS_KEY_ID_SECRET_NAME] }}
+          aws-secret-access-key: ${{ secrets[vars.AWS_SECRET_ACCESS_KEY_SECRET_NAME] }}
+          mask-aws-account-id: true
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository }}
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+          terraform_wrapper: false
+      - name: Deploy to venue
+        id: terraform-deploy
+        working-directory: terraform/
+        env:
+          AWS_DEFAULT_REGION: us-west-2
+        run: |
+          export TF_VAR_edl_base_url=${{ secrets.EDL_BASE_URL }}
+          export TF_VAR_edl_client_id=${{ secrets.EDL_CLIENT_IT }}
+          export TF_VAR_edl_client_secret=${{ secrets.EDL_CLIENT_SECRET }}
+          export TF_VAR_session_encryption_key=${{ secrets.SESSION_ENCRYPTION_KEY }}
+          export TF_VAR_ingest_aws_account=${{ secrets.INGEST_AWS_ACCOUNT }}
+          export TF_VAR_ingest_aws_role=${{ secrets.INGEST_AWS_ROLE }}
+          ./bin/deploy.sh ${{ vars.TF_VENUE }}
```

Comment:
This is my bad - for auto deployment scripts, we use a different command - https://github.com/podaac/swodlr-raster-create/blob/b53c078900b47fd71878a13e3915a55714208b27/.github/workflows/build.yml#L95-L96